src/System.IO.Compression/src/System/IO/Compression/ZipBlocks.cs

```diff
@@ -429,6 +429,95 @@ public static bool TrySkipBlock(BinaryReader reader)
 
             return true;
         }
+
+        public static bool TryValidateBlock(BinaryReader reader, ZipArchiveEntry entry)
+        {
+            const int OffsetToFilename = 26; // from the point after the signature
+
+            if (reader.ReadUInt32() != SignatureConstant)
+                return false;
+
+            if (reader.BaseStream.Length < reader.BaseStream.Position + OffsetToFilename)
+                return false;
+
+            uint minimumVersion = reader.ReadUInt16();
+            uint dataDescriptorBit = reader.ReadUInt16() & (uint)ZipArchiveEntry.BitFlagValues.DataDescriptor;
+            reader.BaseStream.Seek(10, SeekOrigin.Current); // skipping bytes used for Compression method (2 bytes), File last modification time and date (4 bytes) and CRC (4 bytes)
+            long compressedSize = reader.ReadUInt32();
+            long uncompressedSize = reader.ReadUInt32();
+            int filenameLength = reader.ReadUInt16();
+            uint extraFieldLength = reader.ReadUInt16();
+
+            if (reader.BaseStream.Length < reader.BaseStream.Position + filenameLength + extraFieldLength)
+                return false;
+
+            reader.BaseStream.Seek(filenameLength, SeekOrigin.Current);  // skipping Filename
+            long endExtraFields = reader.BaseStream.Position + extraFieldLength;
+            
+            if (dataDescriptorBit == 0)
+            {
+                bool isUncompressedSizeInZip64 = uncompressedSize == ZipHelper.Mask32Bit;
+                bool isCompressedSizeInZip64 = compressedSize == ZipHelper.Mask32Bit;
+                Zip64ExtraField zip64;
+
+                // Ideally we should also check if the minimunVersion is 64 bit or above, but there could zip files for which this version is not set correctly
+                if (isUncompressedSizeInZip64 || isCompressedSizeInZip64)
+                {
+                    zip64 = Zip64ExtraField.GetJustZip64Block(new SubReadStream(reader.BaseStream, reader.BaseStream.Position, extraFieldLength), isUncompressedSizeInZip64, isCompressedSizeInZip64, false, false);
+
+                    if (zip64.UncompressedSize != null)
+                        uncompressedSize = zip64.UncompressedSize.Value;
+                    if (zip64.CompressedSize != null)
```

Comment:
Style: Our style guidelines only permit unbraced ifs in the top method scope.

> A single line statement block can go without braces but the block must be properly indented on its own line and **must not be nested in other statement blocks that use braces**

(<sub>emphasis mine</sub>)

Personally, I only use them in a argument validation preamble